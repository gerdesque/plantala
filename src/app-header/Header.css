.app-logo {
  height: 4em;
  width: 4em;
  margin-right: 1em;
}
