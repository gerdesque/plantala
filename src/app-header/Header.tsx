import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import InvertColorsIcon from '@material-ui/icons/InvertColors';
import VolumeOffIcon from '@material-ui/icons/VolumeOff';
import VolumeUpIcon from '@material-ui/icons/VolumeUp';
import { makeStyles } from '@material-ui/core/styles';
import logo from '../assets/logo.webp';

interface IHeaderProps {
  setStart: any
  isPlaying: boolean
  setSound: any
  setColorMode: any
}

const useStyles = makeStyles((theme) => ({
  header: {
    backgroundColor: theme.palette.primary.main
  },
  icon: {
    marginRight: theme.spacing(2),
    height: theme.spacing(8),
    width: theme.spacing(8),
    cursor: 'pointer'
  },
  title: {
    flexGrow: 1,
  },
}));

export default function Header({ setStart, isPlaying, setSound, setColorMode }: IHeaderProps) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative" className={classes.header}>
        <Toolbar>
          <img src={logo} className={classes.icon} alt="Plantala logo" onClick={() => setStart()} />
          <Typography variant="h2" className={classes.title}>
            Plantala
          </Typography>
          <IconButton aria-label="invert" onClick={() => setSound(isPlaying)}>
            {isPlaying ? <VolumeUpIcon /> : <VolumeOffIcon />}
          </IconButton>
          <IconButton aria-label="invert" onClick={() => setColorMode()}>
            <InvertColorsIcon />
          </IconButton>
        </Toolbar>
      </AppBar>
    </div>
  )
}
