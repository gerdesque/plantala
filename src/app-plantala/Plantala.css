.plantala {
  height: 100vh;
  width: 100vw;
  display: grid;
  grid-template-rows: auto 1fr auto;
}

@media print {
  html, body { width: 210mm; height: 297mm;}
  footer, img, button, h5 { display: none !important; }
  .MuiAppBar-colorPrimary {box-shadow: none !important;}
  h2::before { content: "Malen. Strahlen. "; }
  h2::after { content: "!"; }
  .MuiContainer-root { max-width: 1600px !important; padding: 0 !important }
}