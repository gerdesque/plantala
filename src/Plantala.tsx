import React from 'react';
import Header from './header/Header';
import './Plantala.css';

function Plantala() {
  return (
    <div className="plantala">
      <Header />
    </div>
  );
}

export default Plantala;
