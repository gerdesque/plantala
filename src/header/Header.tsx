import React from 'react';
import logo from '../logo.png';
import './Header.css';

const Header = () => 
  <div className="app-header">  
    <img src={logo} className="app-logo" alt="logo" />
    <h1>Plantala</h1>
  </div>

export default Header;
