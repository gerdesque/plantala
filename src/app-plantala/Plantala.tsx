import React, { Component } from 'react';
import Header from '../app-header/Header';
import Main from '../app-main/Main';
import Footer from '../app-footer/Footer';
import './Plantala.css';
import { plantItems } from '../plantItems';
import { IPlant } from '../plant/Plant';

export enum Action {
  Start = "Start",
  Select = "Select",
  Done = "Done",
  Again = "Again"
}

function* getNextAction(): IterableIterator<Action> {
  let current = Action.Start;
  while (true) {
    yield current;
    yield Action.Select;
    yield Action.Done;
    yield Action.Again;
  }
}

const nextAction = getNextAction();
nextAction.next();

interface IPlantalaState {
  plants: IPlant[],
  activePlant: IPlant,
  action: Action
}

class Plantala extends Component {
  state: IPlantalaState = {
    // values to be displayed in <Card />
    plants: plantItems,
    // active plant for <Plant />
    activePlant: {} as IPlant,
    // action mode to be displayed in <Main />
    action: Action.Start,
  }

  setAction = () => {
    this.setState({action: nextAction.next().value});
  }

  activatePlant = (selectedPlant:IPlant) => {
    const filteredArray =
      this.state.plants.map(plant => plant === selectedPlant ? { ...plant, active: !plant.active } : plant);
    this.setState({ plants: filteredArray })
  }

  render = () => {
    const { plants, action } = this.state;
    const selectedPlants = plants.filter((plant) => (plant.active === true));

    return (
      <div className="plantala">
        <Header />
        <Main
          plants={plants}
          selectedPlants={selectedPlants}
          action={action}
          setAction={this.setAction}
          setSelectedPlant={this.activatePlant} />
        <Footer
          selectedPlants={selectedPlants} />
      </div>
    );
  }
}

export default Plantala;
