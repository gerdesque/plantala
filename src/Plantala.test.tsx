import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Plantala from './Plantala';
import Header from './header/Header';

describe('Plantala', () => {

  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => wrapper = shallow(<Plantala />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div /> container', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the Header Component', () => {
    expect(wrapper.containsMatchingElement(<Header />)).toEqual(true);
  });
});
