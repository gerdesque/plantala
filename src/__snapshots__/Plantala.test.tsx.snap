// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plantala should render correctly 1`] = `
<div
  className="plantala"
>
  <Header />
</div>
`;
